sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java

```diff
@@ -834,4 +887,12 @@ private static Query makeLatestTimestampQuery(String namespace) {
     }
     return queries;
   }
+
+  /** Builds a RunQueryRequest for the give gqlQuery. */
+  private RunQueryRequest makeGqlRequest(String gqlQuery) {
+    String gqlQueryWithZeroLimit = gqlQuery + " limit 0";
+    GqlQuery gql = GqlQuery.newBuilder().setQueryString(gqlQueryWithZeroLimit)
+        .setAllowLiterals(true).build();
+    return RunQueryRequest.newBuilder().setGqlQuery(gql).build();
+  }
```

Comment:
Suggest adding a test that calls apply() with all possible values set to RVPs, to verify that get() is not called at apply time.